src/scores/continuous.py

```diff
@@ -39,19 +41,38 @@ def mse(fcst, obs, reduce_dims=None, preserve_dims=None, weights=None):
             Otherwise: Returns an object representing the mean squared error,
             reduced along the relevant dimensions and weighted appropriately.
     """
+    as_pandas_series = False
+    both_pandas = False
+    if type(fcst) == pandas.Series:
+        fcst = fcst.to_xarray()
+        as_pandas_series = True
+
+    if type(obs) == pandas.Series:
+        obs = obs.to_xarray()
+        as_pandas_series = True
+        if as_pandas_series == True:
+            both_pandas = True
+
 
     error = fcst - obs
     squared = error * error
-    squared = scores.functions.apply_weights(squared, weights)
+    squared = scores.functions.apply_weights(squared, weights)    
 
     if preserve_dims or reduce_dims:
+        
+
         reduce_dims = scores.utils.gather_dimensions(fcst.dims, obs.dims, reduce_dims, preserve_dims)
 
     if reduce_dims is not None:
         _mse = squared.mean(dim=reduce_dims)
     else:
         _mse = squared.mean()
 
+    if both_pandas:
+        _mse = _mse.to_pandas()
+        if isinstance(_mse, numpy.ndarray):
+            _mse = numpy.float64(_mse)        
```

Comment:
That's a really good question and I don't immediately know the answer. I think the current implementation is unlikely to cause major issues in that circumstance because you're going to get a compatible type rather than an incompatible one. It would, I suppose, be possible to capture the type of the original fcst/obs data and pick the highest precision of those types, but I'm not even sure that's the right thing to do. I'll have a bit more of a think on it. Feel free to contribute a test case if you think there's a situation where the current implementation might cause an issue.